src/main/java/seedu/address/model/contact/Contact.java

```diff
@@ -0,0 +1,114 @@
+package seedu.address.model.contact;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Contact in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Contact {
+
+    // Identity fields
+    private final Name name;
+    private final Phone phone;
+    private final Email email;
+
+    // Data fields
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Contact(Name name, Phone phone, Email email, Set<Tag> tags) {
+        requireAllNonNull(name, phone, email, tags);
+        this.name = name;
+        this.phone = phone;
+        this.email = email;
+        this.tags.addAll(tags);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Phone getPhone() {
+        return phone;
+    }
+
+    public Email getEmail() {
+        return email;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both contacts have the same name.
+     * This defines a weaker notion of equality between two contacts.
+     */
+    public boolean equals(Contact otherContact) {
```

Comment:
this is not what i meant... IF YOU REMOVED `isSameContact` CHANGE `UniqueContactList` ACCORDINGLY

```java
public boolean isSameContact(Contact otherContact) {
  return equals(otherContact);
}
```
please change this method to the snippet above and revert `UniqueContactList`